entry/src/block_component.rs

```diff
@@ -614,6 +648,50 @@ impl VersionedBlockMarker {
             _ => Err(BlockComponentError::UnsupportedVersion { version }),
         }
     }
+
+    /// Determines the size of a VersionedBlockMarker in bytes without fully parsing it.
+    fn get_versioned_marker_size(data: &[u8]) -> Result<usize, BlockComponentError> {
+        const VERSION_SIZE: usize = 2;
```

Comment:
should we add some documentation about why this is 2?

Reply:
or maybe define is as `std::mem::size_of::<u16>()`.  

Even better if we had a `struct Version(u16)`.